dietpi/dietpi-config

```diff
@@ -4190,6 +4190,20 @@ Additional benchmarks:
 
 		fi
 
+		# DuckDNS
+		local duckdns_installed=0
+		local duckdns_menutext='[Not Installed] Install DuckDNS now'
+		if [[ -f '/var/lib/dietpi/dietpi-software/installed/duckdns.sh' ]]; then
+
+			duckdns_installed=1
+			duckdns_menutext='[Offline] Invalid DuckDNS details'
+
+			# Check if DuckDNS url returns OK
+			[[ "$(eval $(grep 'duckdns.org/update' /var/lib/dietpi/dietpi-software/installed/duckdns.sh))" == 'OK' ]] && duckdns_menutext='[Online] Change DuckDNS details'
```

Comment:
@ravenclaw900 
Little not about the CodeFactor error here:
- The left sided double quotes in bash `[[ ]]` is not required, as word splitting doesn't play a role and glob expansion (like allowing `*` and `[0-9]`) is only done on unquoted patterns on the right side of `==`.
- But inside of command substitutions `$(command arguments...)`, word splitting is relevant as with every other command call.
- This is particular case, `eval` will to execute the intended `curl`, as `grep` returns multiple arguments, while eval needs to get everything in a single argument. So the statement needs to be:
    ```
    [[ $(eval "$(grep 'duckdns.org/update' /var/lib/dietpi/dietpi-software/installed/duckdns.sh)") == 'OK' ]]
    ```

But do not bother, I'm adding `dietpi-ddns` after doing another important change to `dietpi-software` from my ToDo.